### PR TITLE
dependency: remove iavl

### PR DESCRIPTION
**Pull request by marbar3778**
## Description

Remove IAVL as a dependency to prevent circular dependencies in ll-app

- This needs to be tested by the app anyways. 

Closes: #128

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -31,9 +32,11 @@ require (
 	github.com/spf13/cobra v1.1.1
 	github.com/spf13/viper v1.7.1
 	github.com/stretchr/testify v1.6.1
-	github.com/tendermint/tendermint v0.34.0
```
Comment (liamsi) [inline go.mod:34]: 👍🏼

File: light/rpc/client_test.go [hunk 2]
```diff
@@ -1,90 +1,76 @@
...
+// 			kp := merkle.KeyPath{}
+// 			kp = kp.AppendKey(key, merkle.KeyEncodingURL)
+// 			return kp, nil
+// 		}))
+// 	c.RegisterOpDecoder("ics23:iavl", testOpDecoder)
+// 	res, err := c.ABCIQuery(context.Background(), "/store/accounts/key", key)
+// 	require.NoError(t, err)
+
+// 	assert.NotNil(t, res)
+// }
```
Comment (liamsi) [inline light/rpc/client_test.go:73]: What is the rationale behind commenting this out vs. deleting the code? 

@evan-forbes can you add an equivalent test into the app repo?

Comment (marbar3778) [reply light/rpc/client_test.go:73]: Wanted to leave it around as an example. 

P.S. The sdk doenst have the notion of a light client, so it first needs to be added (https://github.com/cosmos/cosmos-sdk/issues/6563)

Comment (liamsi) [reply light/rpc/client_test.go:73]: Yeah, I think removing makes sense. We can link to the still existing code in the tendermint repo in an issue (over in the app repo or here). Thanks for the context regarding the light client 🙏🏼

**Review by evan-forbes — APPROVED** [review]
As expected, there's no "github.com/tendermint/tendermint" in the `go mod graph` results :+1: 

While using the replace directive on this branch in the cosmos-sdk fork still produces the same error as before, manually replacing tendermint with lazyledger-core *does* work! Most, but not all, of the tests fail to build, some of which is due to the `PreprocessTxs` method to the ABCI interface not yet being ported to the fork of the cosmos-sdk.

thanks, @marbar3778, for fixing this, and saving me from trying an overly complex fix!

**Review by liamsi — APPROVED** [review]
Thanks for tackling this so quickly Marko! 
LGTM

**Comment by marbar3778** [comment]
Want to enable auto deleting of branches after merges?

**Comment by liamsi** [comment]
I think it is already on. I think this does not impact branches in PRs from forks.

**Pull request merged**